tests/lib/timespec_util/src/main.c

```diff
@@ -329,54 +423,138 @@ ZTEST(timeutil_api, test_timespec_from_timeout)
 
 ZTEST(timeutil_api, test_timespec_to_timeout)
 {
-	ztest_test_skip(); /* Provisionally disabled until #92158 is fixed */
-
 	ARRAY_FOR_EACH(tospecs, i) {
 		const struct tospec *const tspec = &tospecs[i];
 		k_timeout_t actual;
+		struct timespec tick_ts;
+		struct timespec rem = {0};
 
 		if (tspec->saturation == 0) {
+			if (!tspec->negative &&
+			    (timespec_compare(&tspec->tspec, &K_TS_NO_WAIT) != 0) &&
+			    (timespec_compare(&tspec->tspec, &K_TS_FOREVER) != 0)) {
+				__ASSERT(timespec_compare(&tspec->tspec, &K_TS_MIN) > 0,
+					 "timespec: {%lld, %lld} is not greater than K_TS_MIN",
+					 (long long)tspec->tspec.tv_sec,
+					 (long long)tspec->tspec.tv_nsec);
+				__ASSERT(timespec_compare(&tspec->tspec, &K_TS_MAX) < 0,
+					 "timespec: {%lld, %lld} is not less than K_TS_MAX",
+					 (long long)tspec->tspec.tv_sec,
+					 (long long)tspec->tspec.tv_nsec);
+			}
+
 			/* no saturation / exact match */
-			actual = timespec_to_timeout(&tspec->tspec);
+			actual = timespec_to_timeout_rem(&tspec->tspec, &rem);
 			zexpect_equal(actual.ticks, tspec->timeout.ticks,
 				      "%d: {%" PRId64 "} and {%" PRId64
 				      "} are unexpectedly different",
 				      i, (int64_t)actual.ticks, (int64_t)tspec->timeout.ticks);
-			continue;
-		}
-
-		if ((tspec->saturation < 0) ||
-		    (timespec_compare(&tspec->tspec, &k_timeout_limits[0]) < 0)) {
-			/* K_NO_WAIT saturation */
-			actual = timespec_to_timeout(&tspec->tspec);
-			zexpect_equal(actual.ticks, K_NO_WAIT.ticks,
+		} else if (tspec->saturation < 0) {
+			__ASSERT(timespec_compare(&tspec->tspec, &K_TS_MIN) <= 0,
+				 "timespec: {%lld, %lld} is not less than or equal to K_TS_MIN",
+				 (long long)tspec->tspec.tv_sec, (long long)tspec->tspec.tv_nsec);
+
+			/* K_TICK_MIN saturation */
+			actual = timespec_to_timeout_rem(&tspec->tspec, &rem);
+			zexpect_equal(actual.ticks, K_TICK_MIN,
 				      "%d: {%" PRId64 "} and {%" PRId64
 				      "} are unexpectedly different",
-				      i, (int64_t)actual.ticks, (int64_t)K_NO_WAIT.ticks);
-			continue;
-		}
-
-		if ((tspec->saturation > 0) ||
-		    (timespec_compare(&tspec->tspec, &k_timeout_limits[1]) > 0)) {
-			/* K_FOREVER saturation */
-			actual = timespec_to_timeout(&tspec->tspec);
-			zexpect_equal(actual.ticks, K_TICKS_FOREVER,
+				      i, (int64_t)actual.ticks, (int64_t)K_TICK_MIN);
+		} else if (tspec->saturation > 0) {
+			/*
+			 * Careful - the compiler can play tricks!
```

Comment:
what is all this?

Reply:
I wish I knew!

Reply:
After fixing a few obvious bugs in the test code https://github.com/aescolar/zephyr/commit/fc8e3fb44b2c14ce5267d2e131036f25116f93ee this part seems to work fine without this weird workaround.